planner/core/logical_plan_builder.go

```diff
@@ -1560,13 +1572,27 @@ func (b *PlanBuilder) resolveGbyExprs(p LogicalPlan, gby *ast.GroupByClause, fie
 	}
 	for _, item := range gby.Items {
 		resolver.inExpr = false
-		retExpr, _ := item.Expr.Accept(resolver)
+		var retExpr ast.Node
+		inPrepare := (len(b.ctx.GetSessionVars().PreparedParams) == 0)
```

Comment:
Actually `notInPrepare`?

Reply:
Yes!

Reply:
``inPrepare`` should let us know whether or not the logical plan is built by ``PrepareExec.Next()``. However, the current impl. may not work correctly if a prepared statement has no parameters. The updated PR will address the issue.